bofire/surrogates/mlp.py

```diff
@@ -109,40 +125,50 @@ def forward(self, X: Tensor):
     @property
     def num_outputs(self) -> int:
         r"""The number of outputs of the model."""
-        return self.mlps[0].layers[-1].out_features  # type: ignore
+        return self.mlps[0].layers[-2].out_features  # type: ignore
 
 
 def fit_mlp(
     mlp: MLP,
-    dataset: RegressionDataSet,
+    dataset: MLPDataset,
     batch_size: int = 10,
     n_epoches: int = 200,
     lr: float = 1e-4,
     shuffle: bool = True,
     weight_decay: float = 0.0,
+    loss_function: Union[
```

Comment:
@jduerholt there is a type error here. I tried removing the type from the loss function and also trying nn.module. I could try nn.modules.loss unless you have any other suggestions?

Reply:
Puh, no idea, then just set a `type: ignore` behind it ...